atom/common/platform_util_win.cc

```diff
@@ -169,33 +170,298 @@ bool OpenExternal(const GURL& url) {
   return true;
 }
 
+
+// Required COM implementation of IFileOperationProgressSink so we can
+// precheck files before deletion to make sure they can be move to the 
+// Recycle Bin.
+class DelFileProgressSink : public IFileOperationProgressSink {
+  public:
+    DelFileProgressSink();
+    ~DelFileProgressSink(void);
+  private:
```

Comment:
nit: Having all of these methods as private is very strange, technically it works because the code making the calls is NOT in the same compiled lib/dll and it is binding against COM v-table slots, but generally you make all of these public.